src/util/ListUtil.re

```diff
@@ -1,22 +1,31 @@
 let rev_if = (b: bool) => b ? List.rev : Fun.id;
 
-let dedup = xs =>
-  List.fold_right(
-    (x, deduped) => List.mem(x, deduped) ? deduped : [x, ...deduped],
-    xs,
-    [],
-  );
-
 let dedup_f = (f, xs) =>
   List.fold_right(
     (x, deduped) => List.exists(f(x), deduped) ? deduped : [x, ...deduped],
     xs,
     [],
   );
 
+let dedup = xs => dedup_f((==), xs);
+
 let are_duplicates = xs =>
   List.length(List.sort_uniq(compare, xs)) == List.length(xs);
 
+/**
+  Groups elements of a list by a specified key.
+
+ {b Note: The groups are not guaranteed to preserve the order of elements from the original list. }
+ 
+  @param key
+  The key function used to determine the grouping key.
+
+  @param xs
+  The list of elements to be grouped.
+
+  @return
+  A list of tuples where each tuple contains the grouping key and a list of elements that belong to that group.
+*/
```

Comment:
![Generated odoc for ListUtil](https://github.com/user-attachments/assets/88b63594-9c66-4ffd-936f-b3fd3e86eb87)

Gearing up for a future odoc pr. Mostly added this so the ordering is noted.